examples/graphics/point-cloud-simulation.html

```diff
@@ -0,0 +1,209 @@
+
+            // color depends on position of particle
+            outColor = vec4(vertexWorld.y * 0.1, 0.1, vertexWorld.z * 0.1, 1);
+        }
+    </script>
+
+    <script id="fshader" type="x-shader/x-fragment">
+        precision mediump float;
+        varying vec4 outColor;
+
+        void main(void)
+        {
+            // color supplied by vertex shader
+            gl_FragColor = outColor;
+
+            // make point round instead of square - make pixels outside of the circle black, using provided gl_PointCoord
+            vec2 dist = gl_PointCoord.xy - vec2(0.5, 0.5);
+            gl_FragColor.a = 1.0 - smoothstep(0.4, 0.5, sqrt(dot(dist, dist)));
+
+        }
+    </script>
+
+    <!-- The script -->
+    <script>
+        var entity, light, camera;
+        var shader, material, diffuseMap;
+        var canvas = document.getElementById("application-canvas");
+
+        // Create the application and start the update loop
+        var app = new pc.Application(canvas);
+        app.start();
+
+        // Set the canvas to fill the window and automatically change resolution to be the same as the canvas size
+        app.setCanvasFillMode(pc.FILLMODE_FILL_WINDOW);
+        app.setCanvasResolution(pc.RESOLUTION_AUTO);
+
+        window.addEventListener("resize", function () {
+            app.resizeCanvas(canvas.width, canvas.height);
+        });
+
+        // Create an Entity with a camera component
+        this.camera = new pc.Entity();
+        this.camera.addComponent("camera", {
+            clearColor: new pc.Color(0, 0, 0)
+        });
+
+        // Add entity into scene hierarchy
+        app.root.addChild(this.camera);
+
+        // allocate two buffers to store positions of particles
+        this.maxNumPoints = 100000;
+        var visiblePoints = 10000;
+        this.positions = new Float32Array(3 * maxNumPoints);
+        this.oldPositions = new Float32Array(3 * maxNumPoints);
+
+        // generate random positions and old positions within small cube (delta between them represents velocity)
+        for (var i = 0; i < 3 * self.maxNumPoints; i++)
+        {
+            self.positions[i] = Math.random() * 2 - 1;
+            self.oldPositions[i] = self.positions[i] + Math.random() * 0.04 - 0.01;
+        }
+
+        // helper function to update vertex of the mesh
+        function updateMesh(mesh) {
+
+            // Set current positions on mesh - this reallocates vertex buffer if more space is needed to test it.
+            // For best performance, we could preallocate enough space using mesh.Clear.
+            // Also turn off bounding box generation, as we set up large box manually
+            mesh.setPositions(this.positions, 3, visiblePoints);
+            mesh.update(pc.PRIMITIVE_POINTS, false);
+        }
+
+        // Create a mesh with dynamic vertex buffer (index buffer is not needed)
+        this.mesh = new pc.Mesh(this.app.graphicsDevice);
```

Comment:
I'm sensing a copy-paste mistake. 😆  `this.app` -> `app`

Reply:
you know me too well ;) fixing.